src/middlewares/authMiddleware.js

```diff
@@ -0,0 +1,21 @@
+import { jwtService } from "../services/jwt.service.js";
+
+export const authMiddleware = (req, res, next) => {
+  const authorization = req.headers['authorization'] || '';
+  const [,token] = authorization.split(' ');
+
+  if (!authorization || !token) {
+    res.status(401).send('Token is not valid');
+    return;
+  }
+
+  const userData = jwtService.verify(token);
+  console.log(userData);
+
+  if (!userData) {
+    res.status(401).send('Token is not valid');
+    return;
+  }
+
```

Comment:
The middleware should assign the `userData` to the `req` object (e.g., `req.user = userData;`) before calling `next()`. This allows subsequent middleware and route handlers to access the authenticated user's data.